### PR TITLE
feat: Simplified Links

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

The <Link> component is clever enough to handle internal/external locations but has a confusing API. This simplifies it.

## Changes

* Removed `href` in favour of always `to`
* Automatic setting of `target="_blank"` ref protection
* Ability to explicitly state for a url if a reload should be forced (e.g. things that use Django routing)

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/lib/components/LemonButton/LemonButton.stories.tsx [hunk 1]
```diff
@@ -212,7 +212,10 @@ export const WithSideAction = (): JSX.Element => {
 export const AsLinks = (): JSX.Element => {
     return (
         <div className="flex gap-2">
-            <LemonButton href="https://posthog.com">External link with "href"</LemonButton>
+            <LemonButton to="https://posthog.com">External link with "href"</LemonButton>
```
Comment (pauldambra) [inline frontend/src/lib/components/LemonButton/LemonButton.stories.tsx:215]: should the story say what the difference is?

**Review by pauldambra — COMMENTED** [review]
Haven't run it yet, a few nits and comments just from reading the code

File: frontend/src/scenes/PreflightCheck/preflightLogic.tsx [hunk 2]
```diff
@@ -283,6 +283,9 @@ export const preflightLogic = kea<preflightLogicType>([
             const appContext = getAppContext()
             const preflight = appContext?.preflight
             if (preflight) {
+                preflight.available_social_auth_providers = {
```
[the remark this answers is not in the document]
Comment (benjackwhite) [reply frontend/src/scenes/PreflightCheck/preflightLogic.tsx:286]: ooooooo whoops! That was me testing some of the social login buttons. Definitely dont want that left in....

File: frontend/src/scenes/billing/BillingLocked.tsx [hunk 3]
```diff
@@ -36,7 +36,8 @@ export function BillingLocked(): JSX.Element | null {
                     type="primary"
                     center={true}
                     fullWidth
-                    href={billing?.subscription_url}
+                    to={billing?.subscription_url}
+                    toForceReload
```
[the remark this answers is not in the document]
Comment (benjackwhite) [reply frontend/src/scenes/billing/BillingLocked.tsx:40]: Changed to `disableClientSideRouting`

**Review by pauldambra — APPROVED** [review]
👨‍🍳 👌

**Pull request merged**